addons/background/src/BackgroundPanel.js

```diff
@@ -72,11 +72,11 @@ export default class BackgroundPanel extends Component {
       this.channel = addons.getChannel();
     }
 
-    this.state = { backgrounds: [] };
+    this.state = { backgrounds: [], currentBackground: null };
 
     this.channel.on('background-set', backgrounds => {
       this.setState({ backgrounds });
-      const currentBackground = api.getQueryParam('background');
+      const currentBackground = api.getQueryParam('background') || this.state.currentBackground;
```

Comment:
I'd prefer URL to be the single source of truth there. We already persist background changes in URL, let's just stop resetting that on story change:
https://github.com/wuweiweiwu/storybook/blob/12c5af9e513f989e03db87475811b86a4cdd718a/addons/background/src/BackgroundPanel.js#L91

Reply:
I see! Will update :)